### PR TITLE
[Editor] Allow MSDF font rendering for custom fonts, add editor setting to control it.

### LINKED ISSUE #95201: Editor doesn't use MSDF font rendering for GraphEdit if a custom main font is used
### Tested versions

- Reproducible in: 4.3.rc 4359c28fe

### System information

Godot v4.3.rc (4359c28fe) - Fedora Linux 40 (KDE Plasma) - X11 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4090 (nvidia; 555.58.02) - 13th Gen Intel(R) Core(TM) i9-13900K (32 Threads)

### Issue description

The editor doesn't use MSDF font rendering for GraphEdit if a custom main font is used.

The one on the left is sharp, but the one on the right is blurry:

### Default font

Screenshot_20240806_175750

### Custom font (Inter)

Screenshot_20240806_175818

Enabling MSDF on custom editor fonts means we might run into https://github.com/godotengine/godot/issues/61207 though, depending on where the custom font comes from. I suggest enabling this by default and seeing if there's a significant amount of reports about broken fonts in GraphEdit (if so, we can add an editor setting to disable this).

It's strange because there already seems to be code in place for this:

https://github.com/godotengine/godot/blob/eabeafd8c3d3fb346853eb1c590ae54b8aa4cd14/editor/themes/editor_fonts.cpp#L234-L249

cc @bruvzg 

### Steps to reproduce

- Use a custom main font in the Editor Settings. Any font will do, I used [Inter Desktop Regular](https://rsms.me/inter/) here.
- Create a visual shader and a few nodes, and increase the zoom level above 100%.
- Notice how text is blurry.
- Revert to the default main font in the Editor Settings.
- Notice how text is crisp.

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes https://github.com/godotengine/godot/issues/95201

### CODE REVIEW

File: doc/classes/EditorSettings.xml [hunk 1]
```diff
@@ -656,6 +656,9 @@
 			Expanding main editor window content to the title, if supported by [DisplayServer]. See [constant DisplayServer.WINDOW_FLAG_EXTEND_TO_TITLE].
 			Specific to the macOS platform.
 		</member>
+		<member name="interface/editor/font_allow_msdf" type="bool" setter="" getter="">
+			If set to [code]true[/code], MSDF font rendering will be used for visual shader graph editor.
```
Comment (akien-mga) [inline doc/classes/EditorSettings.xml:660]: The setting has a generic name so it seems a bit weird to specifically mention the visual shader graph here. Is that the only editor plugin that would make sure of this being enabled>

Comment (bruvzg) [reply doc/classes/EditorSettings.xml:660]: The font itself is part of the main editor theme, but it is the only part of the editor currently using MSDF font.

**Review by Calinou — APPROVED** [review]
Tested locally, it works as expected.

image

File: doc/classes/EditorSettings.xml (hunk 1 above)
Comment (Calinou) [inline doc/classes/EditorSettings.xml:660]:
```suggestion
			If set to [code]true[/code], MSDF font rendering will be used for the visual shader graph editor. You may need to set this to [code]false[/code] when using a custom main font, as some fonts will look broken due to the use of self-intersecting outlines in their font data. Downloading the font from the font maker's official website as opposed to a service like Google Fonts can help resolve this issue.
```

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**